pkg/daemon/update.go

```diff
@@ -814,21 +814,20 @@ func reconcilable(oldConfig, newConfig *mcfgv1.MachineConfig) (*machineConfigDif
 			return nil, errors.New("ignition Passwd Groups section contains changes")
 		}
 		if !reflect.DeepEqual(oldIgn.Passwd.Users, newIgn.Passwd.Users) {
-			// check if the prior config is empty and that this is the first time running.
-			// if so, the SSHKey from the cluster config and user "core" must be added to machine config.
-			if len(oldIgn.Passwd.Users) > 0 && len(newIgn.Passwd.Users) >= 1 {
-				// there is an update to Users, we must verify that it is ONLY making an acceptable
-				// change to the SSHAuthorizedKeys for the user "core"
-				for _, user := range newIgn.Passwd.Users {
-					if user.Name != coreUserName {
-						return nil, errors.New("ignition passwd user section contains unsupported changes: non-core user")
-					}
-				}
-				glog.Infof("user data to be verified before ssh update: %v", newIgn.Passwd.Users[len(newIgn.Passwd.Users)-1])
-				if err := verifyUserFields(newIgn.Passwd.Users[len(newIgn.Passwd.Users)-1]); err != nil {
-					return nil, err
+			if len(oldIgn.Passwd.Users) > 0 && len(newIgn.Passwd.Users) == 0 {
```

Comment:
Just curious, isn't the `len(oldIgn.Passwd.Users) > 0` check technically redundant now that the rest of the checks are pulled out of the if conditional?

This actually does introduce a behaviour change. Consider the following scenario: you want to give a node a ssh passwordHash via cluster installation manifests. This is fine because ignition would write that password, and so long as you didn't change it, we allowed it because it was a ignition written update and not a MC update.

Where this behaviour changes now is that we no longer allow it for the "initial update" from empty MC to initial config. This is because before, `verifyUserFields` (which doesn't allow anything except a core user and an sshkey) would fall under `if len(oldIgn.Passwd.Users) > 0 && len(newIgn.Passwd.Users) >= 1 {` , and the check gets skipped. Now it always gets verified and fails, so your nodes don't install properly (mcd-firstboot fails).

Now this would have eventually come back to bite you if you ever tried to update sshkeys, so we didn't have a great behaviour anyways. The underlying question really is, for ignition supported MC-not supported fields, if you don't change them (e.g. provisioning disks, which is officially supported), what should our behaviour be. Fortunately I don't think we have anywhere documented for other `passwd` sections, so maybe I'm being overly cautious here. If we want to error better that would be good too, because it may be confusing if this ever happened.

Reply:
so right now, at least for OKD, ~core/.ssh/authorized_keys is not created during ignition...  doesn't work in either 4.6 nor 4.7.  There is a key created in ~core/.ssh/authorized_keys.d/

The validation for rendered_master_XXX and rendered_worker_XXX seems to only check for units and files, not Passwd...  99-master-ssh and 99-worker-ssh are in the rendered mc, but never applied

Reply:
@fortinj66 this PR isn't intended for OKD, I think you need to ping on your open issue since OKD has a different way it deals with keys which I believe they are working on

Reply:
OK, thanks...  but it would seem that @sinnykumari sees the same issue that I see with the authorized_keys file not created...

Reply:
yes that was via a bug that i briefly introduced via an older version of this pr that no longer exists.